packages/wallets/src/components/WalletModal/WalletModal.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { ComponentPropsWithoutRef, PropsWithChildren, useRef } from 'react';
+import { useOnClickOutside } from 'usehooks-ts';
+import CloseIcon from '../../public/images/ic-close-dark.svg';
+import { useModal } from '../ModalProvider';
+
+type TWalletModalProps = {
+    should_show_close_icon?: boolean;
+    should_close_on_click_outside?: boolean;
+    onClickCloseIcon?: () => void;
+};
+
+const WalletModal = ({
+    children,
+    onClickCloseIcon,
+    should_show_close_icon,
+    should_close_on_click_outside = true,
```

Comment:
Can we rename the conventional React Modal names like `backdrop_click`
Just a suggestion.

Reply:
would `should_close_on_backdrop_click` be better thou?